app/controllers/web_controller.rb

```diff
@@ -7,12 +7,23 @@ def data
     sanitized_search = I18n.transliterate(params[:q]).gsub(/(\W)/, " ")
     @artist = Search.for(sanitized_search)[0]
 
-    children_artists = @artist.related_artists
 
-    grandchildren_artists = children_artists.map do |child_artist|
-      child_artist.related_artists.delete_if do |grandchild_artist|
-        grandchild_artist.name == @artist.name || children_artists.any? {|a| a.name == grandchild_artist.name}
+    if !@artist
+      begin
+        raise SearchError
+      rescue
+        flash.now[:error] = "Sorry, we couldn't find that artist. Please try again."
       end
+    else
+      @related_artists = @artist.related_artists.sort_by {|artist| artist.followers["total"]}.reverse
```

Comment:
Line is too long. [101/80]<br>Space between { and | missing.<br>Space missing inside }.